### PR TITLE
Write kernels

### PR DESCRIPTION
**Pull request by NPounder**
Write all of the kernels, plus bhr and uncertainties, to geotif.

### CODE REVIEW

File: kafka/modis_kernels_kf.py [hunk 1]
```diff
@@ -138,14 +171,27 @@ def _get_observations_timestep(self, timestep, band=None):
...
-    # stuff stuff stuff
-    drv = gdal.GetDriverByName("GTiff")
-    dst_ds = drv.Create ("tmp/nadir.tif", 2400, 2400, 366, gdal.GDT_Float32,
-                          ['COMPRESS=DEFLATE', 'BIGTIFF=YES', 'PREDICTOR=1',
-                           'TILED=YES'])
-    dst_ds.SetProjection(modis_obs[1].GetProjection())
-    dst_ds.SetGeoTransform(modis_obs[1].GetGeoTransform())
-    
+    # stuff stuff stuff    
+    def create_output_file(template, filename):
```
Comment (jgomezdans) [inline kafka/modis_kernels_kf.py:175]: This should be a general method at the top of the file

File: kafka/modis_kernels_kf.py [hunk 2]
```diff
@@ -138,14 +171,27 @@ def _get_observations_timestep(self, timestep, band=None):
...
+        drv = gdal.GetDriverByName("GTiff")
+        dst_ds = drv.Create (filename, 2400, 2400, 366, gdal.GDT_Float32,
+                              ['COMPRESS=DEFLATE', 'BIGTIFF=YES', 'PREDICTOR=1',
+                               'TILED=YES'])
+        dst_ds.SetProjection(template.GetProjection())
+        dst_ds.SetGeoTransform(template.GetGeoTransform())
+        return dst_ds
+
+    fname_add = '_testing'# ''
+    dst_ds={}
```
Comment (jgomezdans) [inline kafka/modis_kernels_kf.py:185]: These could go into a function, where you specify the output folder too

**Comment by NPounder** [comment]
kernels_comparison_25mayupdate-t288-noprior

BHR for band 1 compared to MCD43 BHR There are a lot of bright spots.

**Comment by NPounder** [comment]
bhrprogression_noprior
But filter is changing the BHR.

I'll put in a pull request for a branch that uses the prior as a starting point and try starting at a better location.

File: kafka/modis_kernels_kf.py [hunk 3]
```diff
@@ -68,13 +68,46 @@ def __init__(self, observations, observation_times,
...
-        x = x_analysis[:(2400*2400)].reshape((2400,2400))
+        x_iso =  x_analysis[:(2400*2400)].reshape((2400,2400))
+        x_vol =  x_analysis[(2400*2400):2*(2400*2400)].reshape((2400,2400))
+        x_geo =  x_analysis[2*(2400*2400):3*(2400*2400)].reshape((2400,2400))
+
+        x_iso_unc =  P_analysis.diagonal()[:(2400*2400)].reshape((2400,2400))
+        x_vol_unc =  P_analysis.diagonal()[(2400*2400):2*(2400*2400)].reshape((2400,2400))
+        x_geo_unc =  P_analysis.diagonal()[2*(2400*2400):3*(2400*2400)].reshape((2400,2400))
+
+        BHR = x_iso+x_vol*0.189184 + x_geo*1.377622
```
Comment (jgomezdans) [inline kafka/modis_kernels_kf.py:79]: Aaahh... see, the weighting factor here is wrong

**Comment by NPounder** [comment]
Thanks for sorting that out!

**Pull request merged**